src/utils.lisp

```diff
@@ -27,11 +27,12 @@
 
 (in-package #:cffi)
 
-(defmacro discard-docstring (body-var &optional force)
-  "Discards the first element of the list in body-var if it's a
-string and the only element (or if FORCE is T)."
-  `(when (and (stringp (car ,body-var)) (or ,force (cdr ,body-var)))
-     (pop ,body-var)))
+(defun omit-docstring (body-var)
+  "Return the original elements of BODY-VAR list but
+without the first element if the type of this element is the string."
+  (if (stringp (first body-var))
+      (rest body-var)
+      body-var))
```

Comment:
`discard-docstring` doesn't destroy anything, it simply points `body-var` elsewhere.

Reply:
also if body has only one element, then string is returned value, not a docstring (at least in Common Lisp), so function name should be omit-car-string or something in this spirit

```
(defun xxx () "HI!") ;-> XXX
(defun yyy () "HI!" "BYE") ;-> YYY
(xxx) ;-> "HI!")
(yyy) ;-> "BYE"
(documentation 'xxx 'function) ;-> NIL
(documentation 'yyy 'function) ;-> "HI!"
```

Reply:
> discard-docstring doesn't destroy anything

Why?  'discard-docstring' expands into (pop ,body-var) and from Hyperspec:
> The effect of (pop place) is roughly equivalent to
>  (prog1 (car place) (setf place (cdr place)))

So, body-var is modified which is forbidden for macro arguments.

> function name should be omit-car-string or something in this spirit

Yeah, I agree with you, maybe "omit-first-string"?

Reply:
`(let ((x (list 1 2))) (setf x (rest x)))` doesn't modify the list `(1 2)`, it points `x` to the second cons cell. The first cons cell remains intact.

Reply:
I would use parse-declarations:parse-body instead of coming with your own (https://common-lisp.net/project/parse-declarations/manual/parse-declarations.html#parse_002dbody).

Reply:
@luismbo Ok, I understand my mistake. So, in the end, I've just changed the signature of the function(change return value from void to list).
I think, that @dkochmanski is right and 'omit-first-string' is a better name.
@sionescu I only changed the function that was used before. I don't sure that we need to use a separate library for this.
What are significant problems with this commit? Is it useless?